modules/core/keeper/keeper.go

```diff
@@ -1,11 +1,16 @@
 package keeper
 
 import (
+	"fmt"
+	"reflect"
+
 	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	capabilitykeeper "github.com/cosmos/cosmos-sdk/x/capability/keeper"
 	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
 
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	upgradekeeper "github.com/cosmos/cosmos-sdk/x/upgrade/keeper"
```

Comment:
Thanks for taking the time to put together this PR.

This puts a hard dependency on the concrete implementations of the sdk staking and upgrade keepers. 
We generally use [expected keepers](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/types/expected_keepers.go#L11-L26) to abstract these kinds of dependencies with a flexible interface.
For example, one may choose to use a custom staking module, that differs slightly from the standard sdk implementation but still honours the interfaces we depend on.

Unsure how others feel about this. cc @colin-axner @seantking @crodriguezvega @AdityaSripal

Reply:
Yeah I think you're right. I will try to make changes accordingly

Reply:
So I have remove the type assertion to sdk-defined keepers and use reflect.ValueOf(stakingKeeper).IsZero(), this should check the value of the interface to see if it's an empty struct or not. What do you think ?